mipengine/controller/node_landscape_aggregator.py

```diff
@@ -114,6 +115,78 @@ def _get_node_socket_addr(node_info: NodeInfo):
     return f"{node_info.ip}:{node_info.port}"
 
 
+class NodeRegistry(BaseModel):
+    def __init__(self, nodes_info=None):
+        if nodes_info:
+            global_nodes = [
+                node_info
+                for node_info in nodes_info
+                if node_info.role == NodeRole.GLOBALNODE
+            ]
+            local_nodes = [
+                node_info
+                for node_info in nodes_info
+                if node_info.role == NodeRole.LOCALNODE
+            ]
+            _nodes_per_id = {
+                node_info.id: node_info for node_info in global_nodes + local_nodes
+            }
+            super().__init__(
+                global_nodes=global_nodes,
+                local_nodes=local_nodes,
+                nodes_per_id=_nodes_per_id,
+            )
+        else:
+            super().__init__()
```

Comment:
https://stackoverflow.com/questions/66932842/constructor-and-pydantic

Reply:
the pydantic way to do this would be to use validators
https://pydantic-docs.helpmanual.io/usage/validators/
what you re doing is a workaround, when the pydantic developers have already worked a much cleaner solution..